src/js/game/hud/parts/building_placer_logic.js

```diff
@@ -114,6 +120,38 @@ export class HUDBuildingPlacerLogic extends BaseHUDPart {
         this.root.camera.upPostHandler.add(this.onMouseUp, this);
     }
 
+    /**
+     * Returns the current base rotation for the current meta-building.
+     * @returns {number}
+     */
+    get currentBaseRotation() {
+        if (!this.root.app.settings.getAllSettings().rotationByBuilding) {
+            return this.currentBaseRotationGeneral;
+        }
+        const building = this.currentMetaBuilding.get();
+        if (building != undefined && this.preferredBaseRotations.hasOwnProperty(building.getId())) {
```

Comment:
I think it would be consistent to just use `if (building && ...)` since this is how its done in the rest of the codebase.

It'd also be nice if you'd use `metaBuilding` as a variable name for it since thats the convention

Reply:
Implemented both of these in both functions.